Reject events larger than the max size allowed

xocasdashdash opened the pull request:
<!-- Please refer to the contribution guidelines before raising a PR: https://github.com/localstack/localstack/blob/master/docs/CONTRIBUTING.md -->

<!-- Why am I raising this PR? Add context such as related issues, PRs, or documentation. -->
## Motivation
To close #11622 


<!-- What changes does this PR make? How does LocalStack behave differently now? -->
## Changes
Adds a new check on the validation of events sent to an event bus.

<!-- Optional section: How to test these changes? -->
<!--
## Testing
Start localstack, configure an event bus and try to send a large payload

-->

<!-- Optional section: What's left to do before it can be merged? -->

## TODO

What's left to do:
- I think i'm missing some bits about the tests, would love to land this PR

xocasdashdash commented:
I have read the CLA Document and I hereby sign the CLA

xocasdashdash commented:
recheck

maxhoheiser commented:
Hey @xocasdashdash great addition thanks.
Two remarks:
- why would you put the test into archive and replay - it is a general feature - it would make more sense to put it into test_events
- you need to install ruff and run make format to resolve formatting issues

xocasdashdash commented:
👋 
- I've fixed the format issues
- I think the test is in the right place now (test_events.py)

maxhoheiser commented:
@xocasdashdash - hey I fixed the test created the snapshot and fixed the validation - since AWS throws a validation error - this is not always so easy to guess how AWS deals with validation - this is why the snapshots help a lot.
Feel free to merge once the pipeline is green.

xocasdashdash commented:
> @xocasdashdash - hey I fixed the test created the snapshot and fixed the validation - since AWS throws a validation error - this is not always so easy to guess how AWS deals with validation - this is why the snapshots help a lot. Feel free to merge once the pipeline is green.

Ahh fantastic, yeah i thought about this and had some local code working in the same way. 

It seems that some specific labels need to be added. Could you add them?

xocasdashdash commented:
@joe4dev @maxhoheiser can you take a look? i had to rebase with some changes that were merged ahead of me

xocasdashdash commented on localstack-core/localstack/services/events/provider.py:
```diff
@@ -199,6 +199,8 @@ def validate_event(event: PutEventsRequestEntry) -> None | PutEventsResultEntry:
             "ErrorCode": "InvalidArgument",
             "ErrorMessage": "Parameter Detail is not valid. Reason: Detail is a required argument.",
         }
+    elif event.get("Detail") and len(event["Detail"]) >= 262144:
```
We can assume that the detail exists here of it would have gone through line 197
```suggestion
    elif len(event["Detail"]) >= 262144:
```

xocasdashdash commented:
@joe4dev Hi!

Sorry for the tag, i've just merged master into the branch and I believe this is ready to be merged. Can you take a look? not sure if there's any next steps i should be following

maxhoheiser reviewed (approved):
@xocasdashdash Great addition - all the tests are green now - feel free to merge :)

xocasdashdash commented:
> @xocasdashdash Great addition - all the tests are green now - feel free to merge :)

hey! i'm not able to! but i'd love to see this change merged! :D
